src/AutoRest.CSharp/Common/Generation/Writers/JsonCodeWriterExtensions.cs

```diff
@@ -304,11 +304,12 @@ private static void DeserializeIntoObjectProperties(this CodeWriter writer, IEnu
                              property.ValueType?.Equals(typeof(string)) !=
                              true) //https://github.com/Azure/autorest.csharp/issues/922
                     {
-                        if (Configuration.AzureArm && property.ValueType?.Equals(typeof(Uri)) == true)
+                        if ((!Configuration.AzureArm || property.ValueType?.Equals(typeof(Uri)) == true)
+                            && !property.IsFlattenedProperty) // if flatten property, then we should not try to assign null/default to parent property
                         {
                             using (writer.Scope($"if ({itemVariable}.Value.ValueKind == {typeof(JsonValueKind)}.Null)"))
                             {
-                                writer.Line($"{propertyVariables[property].Declaration} = null;");
+                                writer.Line($"{propertyVariables[property].Declaration} = default;");
```

Comment:
is it possible that we do not change this? Or add a check if the property is reference type we still use null instead of default to reduce the amount of changed code?